src/bind.c

```diff
@@ -77,18 +77,22 @@ static SEXP vec_rbind(SEXP xs,
     Rf_errorcall(R_NilValue, "Can't bind objects that are not coercible to a data frame.");
   }
 
-  SEXP nms = PROTECT_N(r_names(xs), &n_prot);
-  bool has_names = nms != R_NilValue;
+  bool assign_names = !Rf_inherits(name_spec, "rlang_zap");
+
   bool has_names_to = names_to != R_NilValue;
   R_len_t names_to_loc = 0;
 
   if (has_names_to) {
+    if (!assign_names) {
+      r_abort("Can't zap outer names when `.names_to` is supplied.");
```

Comment:
Is something backwards here? `?vec_rbind` shows that `.names_to` is the argument that accepts `zap()`, not `.name_spec`.

Reply:
Oh now I see what `.name_spec` can also be `zap()`. I had to go hunting in the `?vec_c` docs to see that (but `vec_rbind` does link to `vec_c` so i think that is ok)